molter/help.py

```diff
@@ -0,0 +1,197 @@
+        cmds = await self._gather(ctx)
+
+        if cmd := cmds.get(cmd_name.lower()):
+            _temp = self._generate_command_string(cmd, ctx)
+            _temp += f"\n{cmd.help}"
+            await ctx.reply(self._sanitise_mentions(_temp))
+        else:
+            await ctx.reply(self.not_found_message.format(cmd_name=cmd_name))
+
+    async def _gather(self, ctx: dis_snek.MessageContext | None = None) -> dict[str, molter.MolterCommand]:
+        """
+        Gather commands based on the rules set out in the class attributes.
+
+        args:
+            ctx: The context to use to establish usability.
+
+        returns:
+            dict[str, MolterCommand]: A list of commands fit the class attribute configuration.
+        """
+        out: dict[str, molter.MolterCommand] = {}
+
+        for cmd in self._client.commands.values():
+            cmd: molter.MolterCommand
+
+            if not cmd.enabled and not self.show_disabled:
+                continue
+
+            if cmd == self.cmd and not self.show_self:
+                continue
+
+            elif cmd.hidden and not self.show_hidden:
+                continue
+
+            if ctx and cmd.checks and not self.run_checks:
+                # cmd._can_run would check the cooldowns, we don't want that so manual calling is required
+                for _c in cmd.checks:
+                    if not await _c(ctx):
+                        continue
+
+                if cmd.scale and cmd.scale.scale_checks:
+                    for _c in cmd.scale.scale_checks:
+                        if not await _c(ctx):
+                            continue
+
+            out[cmd.qualified_name] = cmd
+
+        return out
+
+    def _sanitise_mentions(self, text: str) -> str:
+        """
+        Replace mentions with a format that won't ping or look weird in code blocks.
+
+        args:
+            The text to sanitise.
+        """
+        mappings = {
+            "@everyone": "@\u200beveryone",
+            "@here": "@\u200bhere",
+            f"<@{self._client.user.id}>": f"@{self._client.user.username}",
+            f"<@!{self._client.user.id}>": f"@{self._client.user.username}",
+        }
+        for source, target in mappings.items():
+            text = text.replace(source, target)
+
+        return text
+
+    def _generate_command_string(self, cmd: molter.MolterCommand, ctx: dis_snek.MessageContext) -> str:
+        """
+        Generate a string based on a command, class attributes, and the context.
+
+        args:
+            cmd: The command in question.
+            ctx: The context for this command.
+        """
+        _temp = f"`{ctx.prefix if self.show_prefix else ''}{cmd.qualified_name}`"
+
+        if cmd.aliases and self.show_aliases:
+            _temp += "|" + "|".join([f"`{a}`" for a in cmd.aliases])
+
+        if cmd.params and self.show_params:
+            for param in cmd.params:
```

Comment:
You can replace this with a simple `MolterCommand.usage`, which does this and more by default. It *also* allows a dev to make their own usage string if they don't like the default one, which is neat.

Reply:
![image](https://user-images.githubusercontent.com/22540825/161903962-cf1084fe-b98e-4210-a773-fbd707bcc457.png)
I assume that was added outside of `main` which this was origionally based on?

Reply:
It was, resolved.